hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -182,8 +211,10 @@ public void destroyMapStores() {
     public void flushMaps() {
         for (PartitionContainer partitionContainer : partitionContainers) {
             for (String mapName : mapContainers.keySet()) {
-                RecordStore recordStore = partitionContainer.getRecordStore(mapName);
-                recordStore.flush();
+                RecordStore recordStore = partitionContainer.getExistingRecordStore(mapName);
```

Comment:
Rename it to 'findRecordStore'. If you are looking for something, it could be that you can't find it and therefor null is something to be exepcted.